File: backend/src/main/java/lan/dk/podcastserver/service/ByteRangeResourceHandler.java
package lan.dk.podcastserver.service;

import lombok.extern.slf4j.Slf4j;
import org.springframework.core.io.FileSystemResource;
import org.springframework.core.io.Resource;
import org.springframework.stereotype.Component;
import org.springframework.web.servlet.resource.ResourceHttpRequestHandler;

import javax.servlet.http.HttpServletRequest;
import java.io.IOException;
import java.nio.file.Path;

import static io.vavr.API.Option;

/**
 * Created by kevin on 01/09/2017
 */
@Slf4j
@Component
public class ByteRangeResourceHandler extends ResourceHttpRequestHandler {

    public final static String ATTR_FILE = ByteRangeResourceHandler.class.getName() + ".file";

    @Override
    protected Resource getResource(HttpServletRequest request) throws IOException {
        return Option(request.getAttribute(ATTR_FILE))
                .map(Path.class::cast)
                .map(Path::toFile)
                .map(FileSystemResource::new)
                .getOrElseThrow(() -> new RuntimeException("Error during serving of byte range resources"));
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/service/ByteRangeResourceHandler.kt
package com.github.davinkevin.podcastserver.service

import org.springframework.stereotype.Component

/**
 * Created by kevin on 01/09/2017
 */
@Component
class ByteRangeResourceHandler /*: ResourceHttpRequestHandler*/() {

//    public override fun getResource(request: HttpServletRequest): Resource =
//            Option.just(request.getAttribute(ATTR_FILE))
//                    .filter { it is Path }
//                    .map { it as Path }
//                    .map { it.toFile() }
//                    .map { FileSystemResource(it) }
//                    .getOrElse { throw RuntimeException("Error during serving of byte range resources") }

    companion object {
        val ATTR_FILE = ByteRangeResourceHandler::class.java.name + ".file"
    }
}
